### PR TITLE
Data preservation

### PR DESCRIPTION
**Pull request by rebaz36**
## in this milestone we've achieved the followings:

- update rubocop configuration
- create json handler file
- add person json handler
- add book json handler
- add rental json handler
- update student class
- update teacher class
- update rental class
- add json files for store
- fix rubocop errors

### Note/ this project was completed in Pair-Programming with my teammate @JonahKayizzi

### CODE REVIEW

File: src/JSONdata/handle_json.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  # READ PEOPLE
+  def read_people
+    parse_file = read_json('JSONdata/people.json')
+    parse_file.map do |person|
+      if person['type'] == 'Teacher'
+        teacher = Teacher.new(person['age'], person['name'], person['specialization'])
+        teacher.id = person['id']
+        @list_people.push(teacher)
+      else
```
Comment (sja-thedude) [inline src/JSONdata/handle_json.rb:83]: - [ ] Kindly make sure when you create a person, book or rental it should preserve the data in the respective JSON file as explained in the loom video [here](https://www.loom.com/share/c8e59a4aa0204724bad9635d74903b5f).

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @rebaz36, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/JSONdata/handle_json.rb (hunk 1 above)
Comment (rebaz36) [reply src/JSONdata/handle_json.rb:83]: hey @sja-thedude actually the requirements states that 

- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)

so data are saved on exit not instantly when entered

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Good pull request message ✔️
- Good project structure ✔️
209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by rebaz36** [comment]
thanks @uwadonat

**Pull request merged**